lib/modal/edit_product_modal.dart

```diff
@@ -0,0 +1,358 @@
+                      dropdownSearchDecoration: InputDecoration(
+                        labelText: "Category",
+                      ),
+                    ),
+                    popupProps: const PopupProps.menu(
+                      fit: FlexFit.loose,
+                    ),
+                    validator: (value) {
+                      if (value == null || value.isEmpty) {
+                        return '';
+                      }
+
+                      if (errors.containsPartialKey('category')) {
+                        return errors[errors.getValue('category')][0];
+                      }
+                      return null;
+                    },
+                  ),
+                  const SizedBox(
+                    height: 10,
+                  ),
+                  TextFormField(
+                    decoration: const InputDecoration(
+                      hintText: 'Content',
+                      labelText: 'Content',
+                    ),
+                    controller: contentController,
+                    keyboardType: TextInputType.number,
+                    validator: (value) {
+                      if (value == null || value.isEmpty) {
+                        return '';
+                      }
+                      // if value is above 65536
+                      if (double.tryParse(value) == null ||
+                          double.tryParse(value)! > 65535) {
+                        return 'Content must be under 65535';
+                      }
+
+                      if (errors.containsPartialKey('content')) {
+                        return errors[errors.getValue('content')][0];
+                      }
+                      return null;
+                    },
+                  ),
+                  const SizedBox(
+                    height: 10,
+                  ),
+                  DropdownSearch(
+                    asyncItems: (String filter) async {
+                      return api.fetchUnits();
+                    },
+                    selectedItem: product['unit'],
+                    onChanged: (value) {
+                      unitController.text = value.id.toString();
+                    },
+                    dropdownDecoratorProps: const DropDownDecoratorProps(
+                      dropdownSearchDecoration: InputDecoration(
+                        labelText: "Unit",
+                      ),
+                    ),
+                    popupProps: const PopupProps.menu(
+                      fit: FlexFit.loose,
+                    ),
+                    validator: (value) {
+                      if (value == null || value.isEmpty) {
+                        return '';
+                      }
+
+                      if (errors.containsPartialKey('unit')) {
+                        return errors[errors.getValue('unit')][0];
+                      }
+                      return null;
+                    },
+                  ),
```

Comment:
Shows unit abbreviation on load, but shows abbreviation name in dropdown and on change

Reply:
Fixed, name is now in the pre-select also.